test/e2e/windows/dns.go

```diff
@@ -0,0 +1,329 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package windows
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+const dnsTestPodHostName = "dns-querier-1"
+const dnsTestServiceName = "dns-test-service"
+
+var _ = SIGDescribe("DNS", func() {
+
+	ginkgo.BeforeEach(func() {
+		// NOTE(vyta): these tests are Windows specific
+		framework.SkipUnlessNodeOSDistroIs("windows")
+	})
+
+	f := framework.NewDefaultFramework("dns")
+
+	ginkgo.It("should provide DNS for pods for Hostname and Subdomain", func() {
+		// Create a test headless service.
+		ginkgo.By("Creating a test headless service")
+		testServiceSelector := map[string]string{
+			"dns-test-hostname-attribute": "true",
+		}
+		headlessService := framework.CreateServiceSpec(dnsTestServiceName, "", true, testServiceSelector)
+		_, err := f.ClientSet.CoreV1().Services(f.Namespace.Name).Create(headlessService)
+		framework.ExpectNoError(err)
+
+		defer func() {
+			ginkgo.By("deleting the test headless service")
+			defer ginkgo.GinkgoRecover()
+			f.ClientSet.CoreV1().Services(f.Namespace.Name).Delete(headlessService.Name, nil)
+		}()
+
+		hostFQDN := fmt.Sprintf("%s.%s.%s.svc.%s", dnsTestPodHostName, dnsTestServiceName, f.Namespace.Name, framework.TestContext.ClusterDNSDomain)
+		hostNames := []string{hostFQDN, dnsTestPodHostName}
+		namesToResolve := []string{hostFQDN}
+		framework.Logf("hostFQDN: %v", framework.TestContext.ClusterDNSDomain)
+		wheezyProbeCmd, wheezyFileNames := createProbeCommand(namesToResolve, hostNames, "wheezy", f.Namespace.Name, framework.TestContext.ClusterDNSDomain)
+		jessieProbeCmd, jessieFileNames := createProbeCommand(namesToResolve, hostNames, "jessie", f.Namespace.Name, framework.TestContext.ClusterDNSDomain)
+		ginkgo.By("Running these commands on wheezy: " + wheezyProbeCmd + "\n")
+		ginkgo.By("Running these commands on jessie: " + jessieProbeCmd + "\n")
+
+		// Run a pod which probes DNS and exposes the results by HTTP.
+		ginkgo.By("creating a pod to probe DNS")
+		pod1 := createDNSPod(f.Namespace.Name, wheezyProbeCmd, jessieProbeCmd, dnsTestPodHostName, dnsTestServiceName)
+		pod1.ObjectMeta.Labels = testServiceSelector
+
+		validateDNSResults(f, pod1, append(wheezyFileNames, jessieFileNames...))
+	})
+
+	ginkgo.It("should support configurable pod DNS servers", func() {
```

Comment:
shouldn't this also work for Linux? If so, then this test should be added to test/e2e/network/dns.go instead, and ideally, the test would also work on Linux.

Reply:
This is again Windows specific & is in response to #73414

Reply:
Update, while the other test seems to be a duplicate, I don't believe this will work for Linux as it uses ipconfig instead of ifconfig. It is replacing a test that checks /etc/resolv.conf in linux containers

Reply:
Out of curiosity, which is the ``/etc/resolv.conf`` test you're mentioning?

Reply:
[This test](https://github.com/kubernetes/kubernetes/blob/18cc11566f25b11c9fb1f2157a85bb40b410c8a7/test/e2e/network/dns.go#L316) which the "should support configurable pod DNS servers" is based on.